ci/task/compile_models.py

```diff
@@ -0,0 +1,240 @@
+    parser.add_argument("--hf-token", type=str, default=None)
+    parser.add_argument(
+        "--quantization",
+        nargs="+",
+        choices=QUANTIZATION_LIST,
+        default=["q0f16", "q3f16_0", "q3f16_1", "q4f16_0", "q4f16_1", "q4f32_1"],
+    )
+    parser.add_argument("--targets", nargs="+", required=True)
+    return parser.parse_args()
+
+
+def download_model(
+    model_id: str,
+    model_path: Path,
+    hf_token: Optional[str] = None,
+    max_attempts=5,
+):
+    logging.info("Downloading model config from %s", model_id)
+    attempt = 0
+    while attempt < max_attempts:
+        try:
+            attempt += 1
+            snapshot_download(
+                repo_id=model_id,
+                local_dir=model_path,
+                local_dir_use_symlinks=False,
+                token=hf_token,
+                allow_patterns=["config.json"],  # Only download config file
+            )
+            logging.info("Download model config succeeded from %s", model_id)
+            return
+        except Exception as e:  # pylint: disable=broad-except
+            logging.warning(
+                "Download model %s config failed. Attempt number: %d", model_id, max_attempts
+            )
+            if attempt >= max_attempts:
+                raise e
+
+
+def get_target_file_name(target):
+    for pattern, value in TARGET_FILE_NAMES.items():
+        if re.search(pattern, target):
+            return value
+    raise ValueError(f"Unsupported target, expected target with regex {TARGET_FILE_NAMES.keys()}")
+
+
+def gen_config(config: CompileConfig) -> Optional[Exception]:
+    model_name = config.model_name
+    quantization = config.quantization
+    # cmd example:
+    # mlc_chat gen_config dist/models/Llama-2-7b-chat-hf/ \
+    #     --quantization q0f16 \
+    #     --conv-template test-dummy \
+    #     -o dist/libs/Llama-2-7b-chat-hf-q0f16/
+    cmd = [*MLC_CHAT, "gen_config", str(config.model_path)]
+    cmd += ["--quantization", quantization]
+    cmd += ["--conv-template", "test-dummy"]
+    cmd += ["-o", str(config.get_output_path())]
+    result = call_subprocess(cmd)
+    if result:
+        logging.error("Generate MLC config failed with cmd %s: \n%s", result.cmd, result.stdout)
+    else:
+        logging.info("Generated MLC config for %s with quantization %s", model_name, quantization)
+    return result
+
+
+def compile_mode(config: CompileConfig) -> Optional[Exception]:
+    output_path = config.get_output_path()
+    target = config.target
+    output_file = output_path / get_target_file_name(target)
+
+    logging.info("Compiling model %s", config)
+    # cmd example:
+    # mlc_chat compile dist/libs/Llama-2-7b-chat-hf-q0f16/ \
+    #     -o dist/libs/Llama-2-7b-chat-hf-q0f16-cuda.so/ \
+    #     --device cuda
+    cmd = [*MLC_CHAT, "compile", str(output_path)]
+    cmd += ["-o", str(output_file)]
+    cmd += ["--device", str(target)]
+    if target.startswith("cuda"):
+        # Disable cublas for now.
```

Comment:
Let's explicitly add a "TODO" here. We had to temporarily disable cublas in TVM Unity pip wheel due to some unknown breaking in nightly. See [this commit](https://github.com/mlc-ai/package/commit/f5e9e22984c9ad7182f0e4960c7e4204bf71ab38) for more details. CC: @tqchen